### ISSUE TITLE
Scroll To Text Fragment (#:~:text) not working / inconsistent behavior | Flickering on scrolled page reload

### ISSUE THREAD

**Issue by Megarsh** [inertiajs/inertia#1933]
### Version:

- `@inertiajs/vue3` version: 1.2.0

### Describe the problem:

Scroll To Text Fragments, which are usually present on SERP top snippets are not working

Default behavior when those snippets are clicked is for the web page to highlight the text and scroll to the highlighted text position.
Currently Inertia.js eats up the hash navigation and the page is opened without scrolling, although that behavior is also inconsistent because in some cases it does scroll down after flickering

https://github.com/user-attachments/assets/f86c7ee2-e33e-4a24-91ba-7561ac90c1f4

From my observations disabling swapComponent on handleInitialPageVisit call fixes this issue as well as an inconsistent scroll behavior + flickering when reloading scrolled pages (although I don't know if that will break other internal stuff)
![image](https://github.com/user-attachments/assets/42443ae1-a297-4e0a-9455-ab96022e749d)

### Steps to reproduce:

This behavior occurs on one of my production projects running Inertia SSR, but here's a minimal reproduction

https://github.com/Megarsh/inertia-issue

**Comment by pascalbaljet** [comment]
This logic was heavily rewritten in v2. Could you check if it's still a problem there? Thanks!

**Comment by Megarsh** [comment]
> This logic was heavily rewritten in v2. Could you check if it's still a problem there? Thanks!

Checked one of my updated production websites. 

As far as I can tell the scroll restoration is fully fixed however the hash of the url (#:~:text=snippet part) is still being removed

**Comment by pascalbaljet** [comment]
I've dived into this and can confirm that it currently doesn't work with Inertia. The problem is that Inertia always scrolls to the top of the page when there is no URL hash. In the case of text fragments, the browser removes the fragment directive before any scripts run, so the hash appears empty. See here:

- https://github.com/WICG/scroll-to-text-fragment/issues/128

**Comment by pascalbaljet** [comment]
Coming back to this, there is a solution! You need to use [SSR](https://inertiajs.com/server-side-rendering). It actually kind of makes sense: With SSR, the rendered text is already in the response and thus accessible by the browser before it removes the fragment from the URL. The hydration doesn't affect this.

**Comment by Megarsh** [comment]
Yeah my bad, forgot to mention the SSR part. With v2 update the scrolling was fixed. Will add to the conversation for anyone encountering this in the future.

With SSR the scroll to text fragment works, however inertia still removes the hash fragment from the URL
